dashboard/app/models/user.rb

```diff
@@ -885,6 +885,20 @@ def script_level_hidden?(script_level)
     end
   end
 
+  # Is the give script hidden for this user (based on the sections that they are in)
+  def script_hidden?(script)
+    return false if try(:teacher?)
+
+    sections = sections_as_student.select {|s| s.deleted_at.nil?}
```

Comment:
I'm surprised that excluding deleted sections isn't implicit in the `sections_as_student` call.  What determines when we need to say `with_deleted` to get deleted entries, and when we'd expect to have to exclude them manually?

Reply:
I think this is code that was copied from elsewhere and written before we had the acts_as_paranoid. I'll go validate, but I suspect I can just rid of the deleted_at check here (and in the place this was copied from)

Reply:
Manually validated that sections_as_student does not return deleted sections, then fixed up both here and script_level_hidden?